tests/test_resource_manager.py

```diff
@@ -0,0 +1,43 @@
+from contextlib import contextmanager
+from pathlib import Path
+from tests.base import TestCase, main # pylint: disable=import-error,no-name-in-module
+
+from pytest import fixture
+
+from ocrd_utils import pushd_popd, initLogging
+import ocrd_utils.constants
+
+_tempdir = None
+
+@contextmanager
+def monkey_patch_temp_xdg():
+    global _tempdir
+    with pushd_popd(tempdir=True) as tempdir:
+        old_config = ocrd_utils.constants.XDG_CONFIG_HOME
+        old_data = ocrd_utils.constants.XDG_DATA_HOME
+        ocrd_utils.constants.XDG_CONFIG_HOME = tempdir
+        ocrd_utils.constants.XDG_DATA_HOME = tempdir
+        from ocrd.resource_manager import OcrdResourceManager
+        yield tempdir, OcrdResourceManager()
+        ocrd_utils.constants.XDG_CONFIG_HOME = old_config
+        ocrd_utils.constants.XDG_DATA_HOME = old_data
```

Comment:
Wow, that's a great pattern – combining context manager and generator like this

Reply:
I think that is just the standard contextmanager pattern, though you could technically use it as a real generator with multiple yields. I use that a lot for cases like this one where you want to do some setup and cleanup in a reusable way.